src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java

```diff
@@ -0,0 +1,398 @@
+
+package edu.ucla.library.iiif.fester.verticles;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import info.freelibrary.iiif.presentation.v3.Manifest;
+import info.freelibrary.iiif.presentation.v3.properties.ViewingDirection;
+import info.freelibrary.iiif.presentation.v3.properties.behaviors.ManifestBehavior;
+import info.freelibrary.iiif.presentation.v3.utils.Manifestor;
+import info.freelibrary.util.FileUtils;
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+import info.freelibrary.util.RegexDirFilter;
+import info.freelibrary.util.StringUtils;
+
+import edu.ucla.library.iiif.fester.Config;
+import edu.ucla.library.iiif.fester.Constants;
+import edu.ucla.library.iiif.fester.ImageInfoLookup;
+import edu.ucla.library.iiif.fester.MessageCodes;
+import edu.ucla.library.iiif.fester.Op;
+import edu.ucla.library.iiif.fester.utils.IDUtils;
+import io.vertx.config.ConfigRetriever;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.file.FileSystem;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.shareddata.LocalMap;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.Timeout;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+/**
+ * Tests run against the ManifestVerticle.
+ */
+@RunWith(VertxUnitRunner.class)
+public class V3ManifestVerticleTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(V3ManifestVerticleTest.class, Constants.MESSAGES);
+
+    private static final String[] TEST_VERTICLES = new String[] { V3ManifestVerticle.class.getName(),
+        ManifestVerticle.class.getName(), FakeS3BucketVerticle.class.getName() };
```

Comment:
Sigh. One day FakeS3BucketVerticle will be refactored away. Looking forward to that day.